src/sdk/pycli/nnicli/nni_client.py

```diff
@@ -92,48 +66,242 @@ def _create_process(cmd):
     return process.returncode
 
-def start_nni(config_file):
-    """start nni experiment with specified configuration file"""
-    cmd = 'nnictl create --config {}'.format(config_file).split(' ')
-    if _create_process(cmd) != 0:
-        raise RuntimeError('Failed to start nni.')
-
-def stop_nni():
-    """stop nni experiment"""
-    cmd = 'nnictl stop'.split(' ')
-    if _create_process(cmd) != 0:
-        raise RuntimeError('Failed to stop nni.')
-
-def version():
-    """return version of nni"""
-    return _nni_rest_get(VERSION_PATH, 'text')
-
-def get_experiment_status():
-    """return experiment status as a dict"""
-    return _nni_rest_get(STATUS_PATH)
-
-def get_experiment_profile():
-    """return experiment profile as a dict"""
-    return _nni_rest_get(EXPERIMENT_PATH)
-
-def get_trial_job(trial_job_id):
-    """return trial job information as a dict"""
-    assert trial_job_id is not None
-    return _nni_rest_get(os.path.join(TRIAL_JOBS_PATH, trial_job_id))
-
-def list_trial_jobs():
-    """return information for all trial jobs as a list"""
-    return _nni_rest_get(TRIAL_JOBS_PATH)
-
-def get_job_statistics():
-    """return trial job statistics information as a dict"""
-    return _nni_rest_get(JOB_STATISTICS_PATH)
-
-def get_job_metrics(trial_job_id=None):
-    """return trial job metrics"""
-    api_path = METRICS_PATH if trial_job_id is None else os.path.join(METRICS_PATH, trial_job_id)
-    return _nni_rest_get(api_path)
-
-def export_data():
-    """return exported information for all trial jobs"""
-    return _nni_rest_get(EXPORT_DATA_PATH)
+def _check_endpoint(endpoint):
+    if endpoint is None:
+        raise AssertionError("This instance hasn't been connect to an experiment.")
+
+class NNIExperiment:
+    def __init__(self):
+        self.endpoint = None
+        self.exp_id = None
+        self.port = None
+
+    def start_experiment(self, config_file, port=None, debug=False):
+        """
+        Start an experiment with specified configuration file and connect to it.
+
+        Parameters
+        ----------
+        config_file: str
+            path to the config file
+        port: int
+            the port of restful server, bigger than 1024
+        debug: boolean
+            set debug mode
+        """
+        if self.endpoint is not None:
+            raise RuntimeError('This instance has been connected to an experiment.')
+        cmd = 'nnictl create --config {}'.format(config_file).split(' ')
+        if port:
+            cmd += '--port {}'.format(port).split(' ')
+        if debug:
+            cmd += ['--debug']
+        if _create_process(cmd) != 0:
+            raise RuntimeError('Failed to start experiment, please check your config.')
+        else:
+            if port:
+                self.port = port
+            else:
+                self.port = 8080
+            self.endpoint = 'http://localhost:{}'.format(self.port)
+            self.exp_id = self.get_experiment_profile()['id']
```

Comment:
suggest to put this part into a dedicated private member function

Reply:
add a new private function `_exec_command`